### PR TITLE
Clear node messages on deletion

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

_What is this about?_

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/api-service/__app__/node/__init__.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def delete(req: func.HttpRequest) -> func.HttpResponse:
             context=request.machine_id,
         )
 
+    NodeMessage.clear_messages(node.agent_id)
```
Comment (ranweiler) [inline src/api-service/__app__/node/__init__.py:76]: Could new messages also arrive in the meantime?

Is there a time later on in the deletion lifecycle where we should clear the messages, instead?

Comment (bmc-msft) [inline src/api-service/__app__/node/__init__.py]: IMO, these should probably be cleared at the Node.delete() level, such that it's called when the service finalizes the delete.

https://github.com/microsoft/onefuzz/blob/46e84545696b1c69466ebaa1941648d085b98c95/src/api-service/__app__/onefuzzlib/pools.py#L318-L320

**Review by bmc-msft — CHANGES_REQUESTED** [review]

File: src/api-service/__app__/onefuzzlib/pools.py [hunk 2]
```diff
@@ -258,10 +258,11 @@ def is_outdated(self) -> bool:
         return self.version != __version__
 
     def send_message(self, message: NodeCommand) -> None:
-        NodeMessage(
-            agent_id=self.machine_id,
-            message=message,
-        ).save()
+        if not self.delete_requested:
```
Comment (bmc-msft) [inline src/api-service/__app__/onefuzzlib/pools.py:261]: We still need to send the messages here.  In the case we're halting the node, we need to tell it to stop processing any tasks immediately and report back that it has done so.

**Review by bmc-msft — CHANGES_REQUESTED** [review]

File: src/api-service/__app__/node/__init__.py (hunk 1 above)
Comment (chkeita) [reply src/api-service/__app__/node/__init__.py]: moved to Node.delete()

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**